tests/test_ble_lifecycle_receive_targets.py

```
@@ -0,0 +1,1645 @@
+    with iface._state_lock:
+        iface.auto_reconnect = True
+        iface._closed = False
+    try:
+        assert BLELifecycleService._handle_disconnect(iface, "busy") is True
+    finally:
+        iface._disconnect_lock.release()
+
+    iface.close()
+
+
+def test_lifecycle_state_helpers_and_finalize_paths(monkeypatch: pytest.MonkeyPatch) -> None:
```

Ruff (0.15.6)

[warning] Too many statements (87 > 50)

(PLR0915)